src/jp/hidetobara/pasteltouch/PastelTouchActivity.java

```diff
@@ -0,0 +1,535 @@
+		if(s == 0) return;
+		
+		_Canvas.setRadius(s);
+		
+		int r = i.getIntExtra(PenChangeActivity.NAME_R, 0);
+		if(r > 0) _IconSize.setImageResource(r);
+    }
+    private void handleSurfaceActivityResult(Intent i)
+    {
+    	String s = i.getStringExtra(PenChangeActivity.NAME_SURFACE);
+    	if(s.equals("dither")) _Canvas.setSurface(CanvasView.SURFACE_DITHER);
+    	else if(s.equals("random")) _Canvas.setSurface(CanvasView.SURFACE_RANDOM);
+    	else _Canvas.setSurface(CanvasView.SURFACE_FLAT);
+    	
+		int r = i.getIntExtra(PenChangeActivity.NAME_R, 0);
+		if(r > 0) _IconSurface.setImageResource(r);    	
+    }
+    
+    private void newBitmap()
+    {
+    	Display disp = getDisplay();
+    	_Bitmap = Bitmap.createBitmap(disp.getWidth(), disp.getHeight(), Config.ARGB_8888);
+    	_Bitmap.eraseColor(Color.WHITE);
+    }
+    
+    private void convertFitBitmap()
+    {
+    	if(_Bitmap == null) return;
+    	Bitmap bitmap = _Bitmap;
+    	try
+    	{
+			Display display = getDisplay();
+			Matrix m = makeFitMatrix(bitmap, display);
+			RectF r = new RectF(0.0f, 0.0f, bitmap.getWidth(), bitmap.getHeight());
+			m.mapRect(r);
+			_Bitmap = Bitmap.createBitmap(bitmap, 0, 0, bitmap.getWidth(), bitmap.getHeight(), m, true)
+					.copy(Config.ARGB_8888, true);			
+    	}
+    	catch(Exception ex)
+    	{
+			dp(ex.getMessage());
+			_Bitmap = null;
+    	}
+    }
+
+    private Matrix makeFitMatrix(Bitmap b, Display d)
+    {
+    	Matrix m = new Matrix();
+    	int bh = b.getHeight();
+    	int bw = b.getWidth();
+
+    	boolean isRotated = false;
+    	if(d.getWidth() < d.getHeight() && b.getWidth() > b.getHeight()) isRotated = true;
+    	if(d.getWidth() > d.getHeight() && b.getWidth() < b.getHeight()) isRotated = true;
+    	if(isRotated)
+    	{
+    		m.postRotate(90.0f);
+    		bh = b.getWidth();
+    		bw = b.getHeight();
+    	}
+		float sw = (float)d.getWidth() / (float)bw;
+		float sh = (float)d.getHeight() / (float)bh;
+		float scale = sw > sh ? sh : sw;
+		if( scale > 1.0f ) scale = 1.0f;
+		m.postScale(scale, scale);
+    	return m;
+    }
+    
+    private void assignBitmap()
+    {
+    	if(_Bitmap == null || _Canvas == null) return;
+
+    	_Canvas.setBitmap(_Bitmap);	//Bitmapの適応
+		_Canvas.setActivity(this);	//activityの登録
+		_Canvas.reinitialize();		//初期化
+		_Canvas.setBitmapFit(); 	//表示をFitさせる
+    }
+    
+    public void registerFileToGallery(File file)
+    {
+    	if( file == null ) return;
+    	
+        String filepath = file.getPath();
+        String filename = file.getName();
+        String suffix   = Utility.getSuffix(file);
+        String mimetype = "";
+        
+        if ((suffix.equals("jpg")) || (suffix.equals("jpeg"))) {
+            mimetype = "image/jpeg";
+        }else if (suffix.equals("gif")) {
+            mimetype = "image/gif";
+        }else if (suffix.equals("png")) {
+            mimetype = "image/png";
+        }
+        
+        if (mimetype != "") {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.